dolphin/task_manager/tasks/task.py

```diff
@@ -36,17 +37,43 @@ def _set_synced_after(f, *a, **k):
         ret = f(*a, **k)
         lock = coordination.Lock(self.storage_id)
         with lock:
-            storage = db.storage_get(self.context, self.storage_id)
-            storage[constants.DB.DEVICE_SYNC_STATUS] = utils.set_bit(
-                storage[constants.DB.DEVICE_SYNC_STATUS],
-                resource_type,
-                constants.SyncStatus.SYNCED)
-            db.storage_update(self.context, self.storage_id, storage)
+            try:
+                storage = db.storage_get(self.context, self.storage_id)
+            except exception.StorageNotFound:
+                LOG.warn('Storage %s not found when set synced'
+                         % self.storage_id)
+            else:
+                storage[constants.DB.DEVICE_SYNC_STATUS] = utils.set_bit(
+                    storage[constants.DB.DEVICE_SYNC_STATUS],
+                    resource_type,
+                    constants.SyncStatus.SYNCED)
+                db.storage_update(self.context, self.storage_id, storage)
         return ret
 
     return _set_synced_after
 
 
+def check_deleted(resource_type):
+
+    @decorator.decorator
+    def _check_deleted(f, *a, **k):
```

Comment:
What are f a k variables?

Reply:
func, args and kwargs, I'll update the name of these paras